Assets/SEE/GraphProviders/GitGraphProvider.cs

```diff
@@ -0,0 +1,282 @@
+        /// </summary>
+        /// <param name="city">To be checked</param>
+        /// <exception cref="ArgumentException">thrown in case <see cref="VCSPath"/>,
+        /// or <see cref="OldRevision"/> or <see cref="NewRevision"/>
+        /// is undefined or does not exist or <paramref name="city"/> is null or is not a DiffCity</exception>
+        protected void CheckArguments(AbstractSEECity city)
+        {
+            if (city == null)
+            {
+                throw new ArgumentException("The given city is null.\n");
+            }
+            else
+            {
+                if (city is DiffCity diffcity)
+                {
+                    OldRevision = diffcity.OldRevision;
+                    NewRevision = diffcity.NewRevision;
+                    VCSPath = diffcity.VCSPath.Path;
+
+                    if (string.IsNullOrEmpty(VCSPath))
+                    {
+                        throw new ArgumentException("Empty VCS Path.\n");
+                    }
+                    if (!Directory.Exists(VCSPath))
+                    {
+                        throw new ArgumentException($"Directory {VCSPath} does not exist.\n");
+                    }
+                    if (string.IsNullOrEmpty(OldRevision))
+                    {
+                        throw new ArgumentException("Empty Old Revision.\n");
+                    }
+                    if (string.IsNullOrEmpty(NewRevision))
+                    {
+                        throw new ArgumentException("Empty New Revision.\n");
+                    }
+                }
+                else
+                {
+                    throw new ArgumentException("To generate Git metrics, the given city should be a DiffCity.\n");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Calculates the number of lines of code added and deleted for each file changed between two commits and adds them as metrics to <paramref name="graph"/>.
+        /// <param name="graph">an existing graph where to add the metrics</param>
+        /// <param name="vcsPath">the path to the VCS containing the two revisions to be compared</param>
+        /// <param name="oldRevision">the older revision that constitutes the baseline of the comparison</param>
+        /// <param name="newRevision">the newer revision against which the <see cref="oldRevision"/> is to be compared</param>
+        protected void AddLineofCodeChurnMetric(Graph graph, string vcsPath, string oldRevision, string newRevision)
+        {
+            using (var repo = new Repository(vcsPath))
+            {
+                var oldCommit = repo.Lookup<Commit>(oldRevision);
+                var newCommit = repo.Lookup<Commit>(newRevision);
+
+                var changes = repo.Diff.Compare<Patch>(oldCommit.Tree, newCommit.Tree);
+
+                foreach (var change in changes)
+                {
+                    foreach (var node in graph.Nodes())
```

Comment:
[Do not use `var`.](https://github.com/uni-bremen-agst/SEE/wiki/ReviewChecklist)